### PR TITLE
MRG: add `--input-moltype` to singlesketch

### PR DESCRIPTION
**Pull request by ctb**
Fixes https://github.com/sourmash-bio/sourmash_plugin_branchwater/issues/462

- [x] update protein test to make sure it's actually working properly
- [x] test for multiple k in a single param string
- [x] check: are we testing multiple param strings?
- [x] test: what happens with translate?
- [x] what is memory usage like?

### CODE REVIEW

**Comment by ctb** [comment]
@bluegenes @mr-eyes anything I'm missing here? I want to add some tests for paranoia's sake, but otherwise this seems fairly straightforward ;)

**Comment by mr-eyes** [comment]
I couldn't understand the purpose of `-c` in `singlesketch`. But I have a local branch somewhere with `-c` to allow multiprocess sketching on a single fasta file that I was planning to PR it soon, would that conflict with the current `-c` in this PR?

**Comment by ctb** [comment]
> I couldn't understand the purpose of `-c` in `singlesketch`. But I have a local branch somewhere with `-c` to allow multiprocess sketching on a single fasta file that I was planning to PR it soon, would that conflict with the current `-c` in this PR?

I actually thought that this already used multithreading, but I see that it doesn't :).

Your PR would be welcome. Are you doing multiprocess or multithread?

File: src/python/tests/test_sketch.py [hunk 1]
```diff
@@ -1241,14 +1247,74 @@ def test_singlesketch_mult_k(runtmp):
...
+    # Verify that two signatures with different k-mer sizes exist
+    assert len(sigs) == 2
+    assert any(sig.minhash.ksize == 21 for sig in sigs)
+    assert any(sig.minhash.ksize == 31 for sig in sigs)
+
+
+def test_singlesketch_explicit_dna(runtmp):
+    """Test single sketching with explicit DNA in name, and zip output"""
+    fa1 = get_test_data("short.fa")
+    output = runtmp.output("short_dna.sig.zip")
```
Comment (ctb) [inline src/python/tests/test_sketch.py:1279]: note: not actually a zip file, currently.

File: src/singlesketch.rs [hunk 2]
```diff
@@ -21,6 +22,8 @@ pub fn singlesketch(
         }
     };
 
+    // CTB need to check for/error on multiple moltypes, or use
+    // param string parsing; switch to using params_vec, please!
```
[the remark this answers is not in the document]
Comment (ctb) [reply src/singlesketch.rs]: oh, I see. Yes, fixed in d17382f.

**Review by bluegenes — APPROVED** [review]
lgtm

**Comment by mr-eyes** [comment]
> I actually thought that singlesketch already used multithreading, but I see that it doesn't :).
> 
> Your PR would be welcome. Are you doing multiprocess or multithread?

Oh, sorry for the confusion! It's multiprocessing, Opens multiple file handlers, and iterates over sequences in a round-robin fashion. Each core will have a separate signature object, and then I merge all sigs at the end.

**Comment by ctb** [comment]
k - interested in hearing more ;).

**Pull request merged**